File: game_client/Assets/Scripts/ResScripts/NPResCommon/PlayerVariable/NPPlayerVariableObj/NPPlayerVariable_CS_BUFF_HAD_ACTIVE_DAY.cs
using System;
using NPEnum;
using ALPackage;


namespace GOE
{
	public class NPPlayerVariable_CS_BUFF_HAD_ACTIVE_DAY : _ANPBasicPlayerVariableObj
	{
	    private long _m_lBufId;

	    public long buffId { get { return _m_lBufId; } }

	    protected NPPlayerVariable_CS_BUFF_HAD_ACTIVE_DAY()
	    {
	    }

	    /******************
	   * 获取条件类型
	   */
	    public override ENPPlayerVariableType variableType { get { return ENPPlayerVariableType.CS_BUFF_HAD_ACTIVE_DAY; } }

	    public override long calPlayerValue(NPVarInfo _variableInfo)
	    {
#if NP_GAME
	        NPPlayerBuffInfo info = NPPlayer.instance.playerBuffComp.lookup(_m_lBufId);
	        if (null == info)
	            return 0L;

	        return info.hadActiveDay;
#else
	        return 0L;
#endif
	    }


	    public static NPPlayerVariable_CS_BUFF_HAD_ACTIVE_DAY readVariable(ALStringReader _reader)
	    {
	        string bufIdS = _reader.readItem('@');
	        if (null == bufIdS)
	        {
	            UnityEngine.Debug.LogError("高级公式——卡牌状态——配置错误 - BUF_T   example: enum:bufId Error Str: " + _reader.srcString);
	            return null;
	        }

            NPPlayerVariable_CS_BUFF_HAD_ACTIVE_DAY variableObj = new NPPlayerVariable_CS_BUFF_HAD_ACTIVE_DAY();

	        try
	        {
	            variableObj._m_lBufId = long.Parse(bufIdS);

	            return variableObj;
	        }
	        catch (Exception)
	        {
	            UnityEngine.Debug.LogError("高级公式——卡牌状态——配置错误 - BUF_T   example: enum:bufId Error Str: " + _reader.srcString);
	            return null;
	        }
	    }
	}
}

File: game_server/GameRes/src/NPGameRes/GameObjs/PlayerVariable/PlayerVariableObj/NPPlayerVariable_CS_BUFF_HAD_ACTIVE_DAY.java
package NPGameRes.GameObjs.PlayerVariable.PlayerVariableObj;

import NPCommon.CommonObj.NPStringReader;
import NPCommon.Log.CommLog;
import NPEnum.ENPPlayerVariableType;
import NPGameRes.GameObjs.PlayerVariable._ANPBasicPlayerVariableObj;

public class NPPlayerVariable_CS_BUFF_HAD_ACTIVE_DAY extends _ANPBasicPlayerVariableObj
{
    private long _m_lBuffId;//buffId

    public long buffId()
    {
        return _m_lBuffId;
    }

    protected NPPlayerVariable_CS_BUFF_HAD_ACTIVE_DAY()
    {
    }

    /******************
     * 获取条件类型
     */
    @Override
    public ENPPlayerVariableType variableType()
    {
        return ENPPlayerVariableType.CS_BUFF_HAD_ACTIVE_DAY;
    }


    public static NPPlayerVariable_CS_BUFF_HAD_ACTIVE_DAY readVariable(NPStringReader _reader)
    {
        NPPlayerVariable_CS_BUFF_HAD_ACTIVE_DAY variableObj = new NPPlayerVariable_CS_BUFF_HAD_ACTIVE_DAY();

        try
        {
            String bufIdS = _reader.readItem('@');

            variableObj._m_lBuffId = Long.parseLong(bufIdS);

            return variableObj;
        } catch (Exception e)
        {
            CommLog.error("读取NPPlayerVariable_CS_BUFF_HAD_ACTIVE_DAY失败，数据格式错误 src:{}", _reader.getSrcString());
            return null;
        }
    }
}
